UICatalog/Scenarios/BackgroundWorkerCollection.cs

```diff
@@ -13,12 +14,25 @@ namespace UICatalog.Scenarios;
 [ScenarioCategory ("Controls")]
 public class BackgroundWorkerCollection : Scenario
 {
-    public override void Run () { Application.Run<OverlappedMain> (); }
+    public override void Init ()
+    {
+        Application.Run<OverlappedMain> ().Dispose ();
+        Application.Top?.Dispose ();
```

Comment:
I don't know why you have `Dispose` twice and why you removed the `Application.Shutdown` which you want to force the users to use.

Reply:
Should be:

```cs
        Application.Run<OverlappedMain> ().Dispose ();
        Application.Shutdown  ();
```

Reply:
I misunderstood your code. I thought you were disposing Application.Top in addition to the view returned from Run because it may be a different view since SetCurrentOverlappedAsTop changed Top. 

Removing shutdown was an error on my part.

Reply:
On your "you want to force users to call Shutdown" comment. 

Do you believe it's ok for apps to not call Application.Shutdown?

Reply:
No, I mean mandatory. It's the right thing to do.

Reply:
I think you're right also call Application.Top?.Dispose (); because it may not be the main. That's ensure not throwing the assertion.

Reply:
> I think you're right also call `Application.Top?.Dispose ();` because it may not be the main. That's ensure not throwing the assertion.

When we get around to supporting any view in Run, and refactoring overlapped, we really need to make the concept of when Application.Top gets set more deterministic. It's scary that neither of us can articulate exactly why Top (and/or Current) has changed when Run returns here. 

But, for now, I'm not overly concerned because the only place it currently happens is in user code where someone is both using "TopLevelOverlapped" and background tasks/threading ... and this Scenario is pretty much it.

Reply:
You are right. But this has a explanation. This can happen when we start a new RunWorker and close the scenario before it's be showing. I prevented them for not open if the `OverlappedTop` is null. I also fixed an issue where when the scenario is already closed, after start a new RunWorker and the async Post was is called after the `MainLoop` is null.
Maybe now the situation of the `Application.Top` isn't actually the `OverlappedTop`isn't now happen. I will remove the `Application.Top?.Dispose` and doing some tests to check it.